src/packages/@ncigdc/components/analysis/SetOperations/SetOperations.js

```diff
@@ -0,0 +1,146 @@
+  withState('hovering', 'setHovering', new Set()),
+  withRouter,
+)(
+  ({
+    setIds,
+    sets,
+    type,
+    selected,
+    setSelected,
+    push,
+    hovering,
+    setHovering,
+    dispatch,
+    CreateSetButton = CreateSetButtonMap[type],
+  }) => {
+    const setData = Object.entries(sets[type]).filter(([setId]) =>
+      setIds.map(x => x.split(':')[1]).includes(setId),
+    );
+
+    const toggle = op => {
+      selected[selected.has(op) ? 'delete' : 'add'](op);
+      setSelected(selected);
+    };
+
+    const CountComponent = countComponents[type];
+
+    const ops = buildOps({ setData, type });
+
+    const selectedFilters = {
+      op: 'or',
+      content: ops.filter(d => selected.has(d.op)).map(op => op.filters),
+    };
+
+    return (
+      <div style={{ padding: 20 }}>
+        <div style={{ fontSize: 20 }}>Set Operations</div>
+        <div>
+          Click on the areas of the Venn diagram to include them in your
+          result set.
+        </div>
+        <Column style={{ marginTop: 10 }}>
+          <Row>
+            <Measure key="bar-chart">
+              {({ width }) =>
+                <div style={{ position: 'relative', width: '40%' }}>
+                  <div style={{ position: 'absolute' }}>
+                    <Venn
+                      width={width}
+                      data={setIds}
+                      ops={ops}
+                      onClick={toggle}
+                      onMouseOver={op => {
+                        hovering.add(op);
+                        setHovering(hovering);
+                      }}
+                      onMouseOut={() => setHovering(new Set())}
+                      getFillColor={(d, i) => {
+                        return hovering.has(d.op)
+                          ? colors[1]
+                          : selected.has(d.op) ? colors[2] : colors[0];
+                      }}
+                    />
+                  </div>
+                  {ops.length === 3
+                    ? <TwoSetOverlay
+                        width={width}
+                        ops={ops}
+                        CountComponent={CountComponent}
+                      />
+                    : <ThreeSetOverlay
+                        width={width}
+                        ops={ops}
+                        CountComponent={CountComponent}
+                      />}
+                </div>}
+            </Measure>
+            <Column style={{ width: '60%' }}>
+              {SetTable({
+                push,
+                setData,
+                type,
+                CountComponent,
+                CreateSetButton,
+              })}
+              <hr />
+              {OpsTable({
+                type,
+                selected,
+                toggle,
+                push,
+                ops,
+                dispatch,
+                CountComponent,
+                selectedFilters,
+                CreateSetButton,
+              })}
+              />
```

Comment:
I don't think this matches a tag? 
![image](https://user-images.githubusercontent.com/3953093/30132467-6da539be-931e-11e7-8e3d-51774f86d33d.png)

Reply:
thanks, removed